### PR TITLE
docs(README): add details on usage and purpose

### PR DESCRIPTION
**Pull request by ecarlisle**
Feedback is welcome on the copy to confirm it is accurate.

### CODE REVIEW

**Comment by ecarlisle** [comment]
Ok, I believe we have achieved clean PR.  :-)

**Comment by michael-ciniawsky** [comment]
I try to review it as soon as possible :)

File: README.md [hunk 1]
```diff
@@ -39,7 +39,11 @@ module.exports = {
 };
 ```
 
-This extracts SourceMaps from all js files (including node_modules). This is not very performant, so you may want to only apply the loader to relevant files.
+`source-map-loader` extracts existing [sourcemaps](http://blog.teamtreehouse.com/introduction-source-maps) from all JavaScript entries.  This includes both inline sourcemaps as well as map data linked via URL. All map data is passed to WebPack for processing as per a chosen [source mapping style](https://webpack.js.org/configuration/devtool/) specified by the `devtool` option in [webpack.conf.js](https://webpack.js.org/configuration/).
+ +
+ +This loader is especially useful when using 3rd-party libraries having their own source maps. If not extracted and processed into the map of the WebPack bundle, browsers may misinterpret map data. `source-map-loader` allows Webpack to maintain map data continuity across libraries so ease of debugging is preserved.
```
Comment (bebraw) [inline README.md]: /s/WebPack/webbpack

Comment (michael-ciniawsky) [inline README.md]: `webbpack` => `webpack` 😛

File: README.md [hunk 2]
```diff
@@ -39,7 +39,11 @@ module.exports = {
 };
 ```
 
-This extracts SourceMaps from all js files (including node_modules). This is not very performant, so you may want to only apply the loader to relevant files.
+`source-map-loader` extracts existing [sourcemaps](http://blog.teamtreehouse.com/introduction-source-maps) from all JavaScript entries.  This includes both inline sourcemaps as well as map data linked via URL. All map data is passed to WebPack for processing as per a chosen [source mapping style](https://webpack.js.org/configuration/devtool/) specified by the `devtool` option in [webpack.conf.js](https://webpack.js.org/configuration/).
```
Comment (michael-ciniawsky) [inline README.md]: Is the course mentioned a free one (available for all) ?

Comment (michael-ciniawsky) [inline README.md]: ` from all JavaScript entries.  <= This includes ` => ` from all JavaScript entries. This includes` (Spacing)

Comment (michael-ciniawsky) [inline README.md]: `WebPack` => `webpack`

File: README.md (hunk 1 above)
Comment (michael-ciniawsky) [inline README.md]: Either use the term `map` or `source map` across the whole text (Consistency)

**Comment by michael-ciniawsky** [comment]
@ecarlisle See the review good after that 👍 , sry for the delay 😊

File: README.md (hunk 2 above)
Comment (ecarlisle) [inline README.md]: @michael-ciniawsky : I'm assuming there was a course mentioned in the teamtreehouse.com link. I removed the link as it's probably unneeded.

**Comment by ecarlisle** [comment]
@michael-ciniawsky , @bebraw - FYI, I believe I've addressed all the comments thus far.
Thanks!

File: README.md [hunk 3]
```diff
@@ -39,7 +39,11 @@ module.exports = {
 };
 ```
 
-This extracts SourceMaps from all js files (including node_modules). This is not very performant, so you may want to only apply the loader to relevant files.
+`source-map-loader` extracts existing source maps from all JavaScript entries. This includes both inline source maps as well as those linked via URL. All source map data is passed to webpack for processing as per a chosen [source map style](https://webpack.js.org/configuration/devtool/) specified by the `devtool` option in [webpack.conf.js](https://webpack.js.org/configuration/).
```
Comment (michael-ciniawsky) [inline README.md]: `webpack.conf.js` => `webpack.config.js` ('The final Typo') 😛

Comment (ecarlisle) [inline README.md]: Haha, all my typos were with the **most** important terms (e.g. getting the proper casing on _webpack_ and getting the config filename name right). :-D

**Comment by ecarlisle** [comment]
Completed that next round of review, the config name typo.
Do you believe the new description is accurate in what the `source-map-loader` does?  Did I leave anything important out?
I really appreciate the teamwork on this!

File: README.md [hunk 4]
```diff
@@ -39,7 +39,7 @@ module.exports = {
 };
 ```
 
-`source-map-loader` extracts existing source maps from all JavaScript entries. This includes both inline source maps as well as those linked via URL. All source map data is passed to webpack for processing as per a chosen [source map style](https://webpack.js.org/configuration/devtool/) specified by the `devtool` option in [webpack.conf.js](https://webpack.js.org/configuration/).
+`source-map-loader` extracts existing source maps from all JavaScript entries. This includes both inline source maps as well as those linked via URL. All source map data is passed to webpack for processing as per a chosen [source map style](https://webpack.js.org/configuration/devtool/) specified by the `devtool` option in [webpack.confis.js](https://webpack.js.org/configuration/).
```
Comment (michael-ciniawsky) [inline README.md]: `webpack.confis.js` => `webpack.config.js`

**Comment by michael-ciniawsky** [comment]
@ecarlisle Nope everything :+1:, besides the small typo in the link :)

File: README.md (hunk 4 above)
Comment (ecarlisle) [inline README.md]: Got that. Uploaded, looked, groaned. Push another commit. :-)

**Comment by ecarlisle** [comment]
@michael-ciniawsky : Cool! Typo link is fixed. :-)

**Pull request merged**